File: server/checkmate/socket/parsers.py
import json

def receiveParser(text_data):
    json_data = json.loads(text_data)
    action = json_data["action"]
    data = json_data["data"]
    
    return [action, data]

def sendParser(action, data):
    json_data = { "action" : action, "data" : data }
    text_data = json.dumps(json_data)
    
    return text_data

File: client/src/api/socket/parsers.ts
import { Action } from "../../types/socket/global"
import { SocketData } from "../../types/socket/receiveData"
import { SendData } from "../../types/socket/sendData"

export const receiveParser = (textData: string): SocketData=>{
    const jsonData: SocketData = JSON.parse(textData)
    const action = jsonData.action
    const data = jsonData.data

    return {action, data}
}


export const sendParser = <D extends SendData>(action: Action, data: D)=>{
    const jsonData = {action, data}
    const textData = JSON.stringify(jsonData)

    return textData
}